File: 01-backend/src/main/java/ee/insa/crmapp/model/Insurer.java
package ee.insa.crmapp.model;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "insurer")
public class Insurer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;
    @OneToOne
    private Firm firm;
    private String name;
    private String invoicePrefix;
    private String comment;
    private String url;
    private String username;
    private String password;

    public Insurer(Firm firm, String name, String invoicePrefix, String comment) {
        this.firm = firm;
        this.name = name;
        this.invoicePrefix = invoicePrefix;
        this.comment = comment;
    }
}

File: 02-frontend/src/app/model/insurer.ts
import {Firm} from "./firm";

export class Insurer {
  public id: number;
  public firm: Firm;
  public name: string;
  public invoicePrefix: string;
  public comment: string;
  public url: string;
  public username: string;
  public password: string;

  constructor() {
    this.id = 0;
    this.firm = null;
    this.name = '';
    this.invoicePrefix = '';
    this.comment = '';
    this.url = '';
    this.username = '';
    this.password = '';
  }
}
